php/libraries/Database.class.inc

```diff
@@ -966,19 +966,27 @@ class Database
      *
      * @param string $glue      The glue to pass to php's implode function
      * @param array  $dataArray The array with keys to implode
+     * @param string $clause    The clause type (set_ or where_) to correctly
+     *                          implement the NULL parameter
      *
      * @return string the string containing the imploded array
      * @access private
      */
-    function _implodeWithKeys($glue, $dataArray)
+    function _implodeWithKeys($glue, $dataArray, $clause = 'set_')
```

Comment:
Since `$clause` seems to be used as a boolean, I'd be in favour of naming it `$isSetClause` with value true for set clauses and false otherwise.

Reply:
There is already a variable used as "set_" elsewhere in that module. So I used it the same way. It is also more clear to a humain than 0 or 1, no need to put comment. Of course binary would be (a little bit) faster at execution